[MTM-46157]-Add-requirements-for-managing-user-hierarchies (#991)
SoftwareAG/c8y-docs: content/users-guide/enterprise-tenant-bundle/user-hierarchies.md

  - /users-guide/enterprise-edition/#user-hierarchies
---

{{< c8y-admon-req >}}
SUBSCRIPTIONS:

> Tenant must be subscribed to the application "feature-user-hierachy".

Reviewer: ```suggestion
The tenant must be subscribed to the application "feature-user-hierachy".
```

APPLICATION ACCESS:

> To access the feature, application access should be granted for user for "Administration" application.

Reviewer: ```suggestion
To access the feature, application access must be granted to the user for the Administration application.
```

ROLES & PERMISSIONS:

"User management" permission type:

> * To view all users: READ permission.

Author: There is some guides we used before, lets keep original approach for now, ref:
content/users-guide/cockpit-bundle/exports.md
content/users-guide/administration-bundle/managing-users.md

> * To view all users: READ permission.

Reviewer: I agree. Let´s keep it consistent. Moreover, I think it is clearer from a user perspective to first state what you want to do and second which permission you need to do so. Here it is quite simple but we have much more complex permissions.

> * To manage all users: ADMIN permission.

Author: There is some guides we used before, lets keep original approach for now, ref:
content/users-guide/cockpit-bundle/exports.md
content/users-guide/administration-bundle/managing-users.md

> * To create users: CREATE permission. By assigning this permission you can create own subusers and manage underlying hierarchy of users.

Reviewer: ```suggestion
* To create users: CREATE permission. By assigning this permission you can create own sub-users and manage the underlying hierarchy of users.
```

> * To create users: CREATE permission. By assigning this permission you can create own sub-users and manage the underlying hierarchy of users.

Author: There is some guides we used before, lets keep original approach for now, ref:
content/users-guide/cockpit-bundle/exports.md
content/users-guide/administration-bundle/managing-users.md

> On tenant creation, there are default roles available that can be used as a sample configuration for the above-mentioned permissions:

Reviewer: ```suggestion
On tenant creation, default roles are available which can be used as a sample configuration for the above-mentioned permissions:
```

* Global User Manager - Can access and modify the full user hierarchy (all users).

> * Shared User Manager - Can create new users as his own subusers and manage underlying users' hierarchy.

Reviewer: ```suggestion
* Shared User Manager - Can create new users as his own sub-users and manage the underlying user hierarchy.
```

{{< /c8y-admon-req >}}

> With user hierarchies you can reflect independent organizational entities in {{< product-c8y-iot >}} that still share the same database. These entities can have limited permissions to subsets of the shared data and can manage their own sub-users.

Reviewer 2: "t independent organizational entities in Cumulocity IoT" sounds a bit weird, maybe:
```suggestion
With user hierarchies you can model the structure of organizational entities in your company which share the same database in {{< product-c8y-iot >}} platform. These entities can have limited permissions to subsets of the shared data and can manage their own sub-users.
```
